testsuites/CBLTester/System_test_multiple_device/conftest.py

```diff
@@ -0,0 +1,290 @@
+
+    parser.addoption("--device", action="store_true",
+                     help="Enable device if you want to run it on device", default=False)
+
+    parser.addoption("--community", action="store_true",
+                     help="If set, community edition will get picked up , default is enterprise", default=False)
+
+    parser.addoption("--sg-ssl",
+                     action="store_true",
+                     help="If set, will enable SSL communication between Sync Gateway and CBL")
+
+
+# This will get called once before the first test that
+# runs with this as input parameters in this file
+# This setup will be called once for all tests in the
+# testsuites/CBLTester/CBL_Functional_tests/ directory
+@pytest.fixture(scope="session")
+def params_from_base_suite_setup(request):
+    liteserv_platforms = request.config.getoption("--liteserv-platforms")
+    liteserv_versions = request.config.getoption("--liteserv-versions")
+    liteserv_hosts = request.config.getoption("--liteserv-hosts")
+    liteserv_ports = request.config.getoption("--liteserv-ports")
+
+    platform_list = liteserv_platforms.split(',')
+    version_list = liteserv_versions.split(',')
+    host_list = liteserv_hosts.split(',')
+    port_list = liteserv_ports.split(',')
+
+    if len(platform_list) != len(version_list) != len(host_list) != len(port_list):
+        raise Exception("Provide equal no. of Parameters for host, port, version and platforms")
+    skip_provisioning = request.config.getoption("--skip-provisioning")
+    sync_gateway_version = request.config.getoption("--sync-gateway-version")
+    mode = request.config.getoption("--mode")
+
+    server_version = request.config.getoption("--server-version")
+    enable_sample_bucket = request.config.getoption("--enable-sample-bucket")
+    xattrs_enabled = request.config.getoption("--xattrs")
+    create_db_per_suite = request.config.getoption("--create-db-per-suite")
+    device_enabled = request.config.getoption("--device")
+    sg_ssl = request.config.getoption("--sg-ssl")
+    resume_cluster = request.config.getoption("--resume-cluster")
+    generator = request.config.getoption("--doc-generator")
+    no_db_delete = request.config.getoption("--no-db-delete")
+
+#     community_enabled = request.config.getoption("--community")
+#
+#     testserver_list = []
+#     for platform, version, host, port in zip(platform_list,
+#                                              version_list,
+#                                              host_list,
+#                                              port_list):
+#         testserver = TestServerFactory.create(platform=platform,
+#                                               version_build=version,
+#                                               host=host,
+#                                               port=port,
+#                                               community_enabled=community_enabled)
+#
+#         log_info("Downloading TestServer ...")
```

Comment:
you need to add logic to download, install and start the app on multiple clients